### ISSUE TITLE
Bug in no-extra-bind

### ISSUE THREAD

**Issue by mjomble** [eslint/eslint#4806]
There is a bug in the symmetry of pushing/popping on the `scope` stack in the code of this rule.
The `pop` is skipped in case of a necessary `bind` which breaks cases like this one:

```javascript
var a = function() {
  (function(){
    (function(){ this.d }.bind(c))
  })
}.bind(b)
```

`bind(b)` is unnecessary, but the error does not get reported.

I already have a fix and created the issue only so I could submit the pull request.

**Comment by eslintbot** [comment]
@mjomble Thanks for the issue! If you're reporting a bug, please be sure to include:

1. The version of ESLint you are using (run `eslint -v`)
2. What you did (the source code and ESLint configuration)
3. The actual ESLint output complete with numbers
4. What you expected to happen instead

Requesting a new rule? Please see [Proposing a New Rule](http://eslint.org/docs/developer-guide/contributing/new-rules) for instructions.
